### PR TITLE
feat(WalletConnect): Implement sign request modal

### PR DESCRIPTION
**Pull request by alexjba**
### What does the PR do
Closes: #14762 

1. Implementing sign request modal based on SignTransactionModalBase
2. Adding storybook page
3. Integrate it in the app
4. Removing DAppRequestModal
5. Update RoundImageWithBadge to preserve aspect ratio between badge and main image

What it's not included:
1. Proper fees calculation. Will be handled in #15192
2. Background sign watermark. Will be handled in #14498

### Affected areas

Wallet connect

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

### How to test

- In storybook using DAppSignRequestModal page and DAppWorkFlow page
- In the app by enabling `FLAG_DAPPS_ENABLED` env variable

### Risk 

Described potential risks and worst case scenarios.

Tick **one**:
- [x] Low risk: 2 devs MUST perform testing as specified above and attach their results as comments to this PR **before** merging.
- [ ] High risk: QA team MUST perform additional testing in the specified affected areas **before** merging.

### CODE REVIEW

**Comment by stefandunca** [comment]
Thanks @alexjba for the great improvements. Just tested the PR and the transaction is broken as you can see in the bellow screenshot.
image

I'm also not sure how do you want to deal with the sign message vs. sign/send transaction? Would you plan to have another `SignMessageModalBase.qml` or execute all in the SignTransactionModalBase? The naming refactoring is a littlebit confusing to me. We had request that generalizes to sign message sign/send transaction but now I see sign transaction

File: ui/app/AppLayouts/Wallet/services/dapps/WalletConnectService.qml [hunk 1]
```diff
@@ -28,6 +28,7 @@ QObject {
 
     readonly property alias dappsModel: dappsProvider.dappsModel
     readonly property alias requestHandler: requestHandler
+    readonly property int loginType: root.store.loginType
```
Comment (stefandunca) [inline ui/app/AppLayouts/Wallet/services/dapps/WalletConnectService.qml:31]: I'm not sure bringing account state to service is a good idea. This is embedded with the account, right? Can't it be extracted from the account?

Comment (stefandunca) [reply ui/app/AppLayouts/Wallet/services/dapps/WalletConnectService.qml:31]: On a second thought `selectedAccountAddress` also looks like an unnecessary state added tot he service interface, why not extract it from the `walletRootStore` directly?

Comment (alexjba) [reply ui/app/AppLayouts/Wallet/services/dapps/WalletConnectService.qml:31]: Ohh, you're right! Thanks! 🍻 
Was looking at this service as the only dependency it needs to communicate with the outside and maybe extract it someday out of the UI component. But IDK I've totally dismissed the idea of adding another input properties in the `DAppsWorkflow`.

Fixing it :D

Comment (alexjba) [reply ui/app/AppLayouts/Wallet/services/dapps/WalletConnectService.qml]: Done! Please have a look here https://github.com/status-im/status-desktop/pull/15520/commits/84053e2b2abaee3fca748eca48519404be2745ad

Sign transaction is fixed as well.

Thanks for this! 👍

File: ui/imports/shared/popups/walletconnect/DAppSignRequestModal.qml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    headerIconComponent: RoundImageWithBadge {
+        imageUrl: root.dappIcon
+        width: 40
+        height: 40
+    }
+
+    gradientColor: Utils.setColorAlpha(root.accountColor, 0.05) // 5% of wallet color
+    headerMainText: qsTr("%1 wants you to sign this %2 with %3")
+                    .arg(root.dappName)
+                    .arg(root.signingTransaction ? "transaction" : "message")
```
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/DAppSignRequestModal.qml:57]: Pfff... :D make it two sentences pls; this is a riddle

File: ui/imports/shared/popups/walletconnect/assets/sign-background.svg
Comment (caybro) [inline ui/imports/shared/popups/walletconnect/assets/sign-background.svg:1]: Where do you use this file?

File: ui/imports/shared/stores/DAppsStore.qml [hunk 3]
```diff
@@ -7,6 +7,8 @@ QObject {
 
     required property var controller
 
+    readonly property var globalUtilsInst: globalUtils
```
Comment (caybro) [inline ui/imports/shared/stores/DAppsStore.qml:10]: `readonly` still works when overridden?

**Review by caybro — COMMENTED** [review]
Looks good, just some minor things (plus what stefand pointed out) :)

File: ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml [hunk 4]
```diff
@@ -11,13 +11,15 @@ import shared.popups.walletconnect 1.0
...
-import shared.stores 1.0
 import utils 1.0
 
 DappsComboBox {
     id: root
 
     required property WalletConnectService wcService
+    // Values mapped to Constants.LoginType
+    property int loginType
+    property string selectedAccountAddress
```
[the remark this answers is not in the document]
Comment (alexjba) [reply ui/app/AppLayouts/Wallet/panels/DAppsWorkflow.qml:22]: `selectedAccountAddress` not really needed. Has a default.
But the `loginType` is indeed needed for it to work as designed.

File: ui/imports/shared/popups/walletconnect/assets/sign-background.svg
Comment (alexjba) [reply ui/imports/shared/popups/walletconnect/assets/sign-background.svg:1]: removed

File: ui/imports/shared/stores/DAppsStore.qml (hunk 3 above)
Comment (alexjba) [reply ui/imports/shared/stores/DAppsStore.qml]: removed as well. I've started with a much larger scope and later discovered we have other tasks to handle fees and the background 😄

**Review by stefandunca — APPROVED** [review]
Thanks for fixing the issues. I'm looking forward to have this merged.

**Review by caybro — APPROVED** [review]
Cool :)

**Pull request merged**